.github/workflows/cicd.yml

```diff
@@ -1,39 +1,107 @@
+# https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
 name: MSBuild
 
 on:
-  push:
-    branches: [ main ]
+  schedule:
+    - cron: '00 21 * * *'
   pull_request:
-    branches: [ main ]
-
-env:
-  # Path to the solution file relative to the root of the project.
-  SOLUTION_FILE_PATH: .
-
-  # Configuration type to build.
-  # You can convert this to a build matrix if you need coverage of multiple configuration types.
-  # https://docs.github.com/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
-  BUILD_CONFIGURATION: Release
-
-permissions:
-  contents: read
 
 jobs:
   build:
-    runs-on: windows-latest
+    strategy:
+      matrix:
+        configurations: [Debug, Release]
+    runs-on: windows-2019
+    env:
+      # Solution file names.
+      EBPF_FOR_WINDOWS_SOLUTION_FILE: ebpf-for-windows.sln
+      DEMO_SOLUTION_FILE: ebpf-for-windows-demo.sln
+
+      # Configuration type to build.
+      # You can convert this to a build matrix if you need coverage of multiple configuration types.
+      # https://docs.github.com/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
+      BUILD_CONFIGURATION: ${{matrix.configurations}}
+
+      BUILD_PLATFORM: x64
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
 
     - name: Add MSBuild to PATH
-      uses: microsoft/setup-msbuild@v1.0.2
+      uses: microsoft/setup-msbuild@becb80cf9a036187bb1e07e74eb64e25850d757a
+
+    - name: Install ProcDump64
+      working-directory: ${{env.GITHUB_WORKSPACE}}
+      run: |
+        curl -fsSL -o Procdump.zip https://download.sysinternals.com/files/Procdump.zip
+        7z x Procdump.zip -y -o"C:/Program Files/ProcDump"
+        echo "C:\Program Files\ProcDump" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+        mkdir c:/dumps/${{env.BUILD_PLATFORM}}/${{env.BUILD_CONFIGURATION}}
+        echo "test" > c:/dumps/${{env.BUILD_PLATFORM}}/${{env.BUILD_CONFIGURATION}}/test.txt
+
+    - name: Set AeDebug registry key
+      working-directory: c:/dumps/${{env.BUILD_PLATFORM}}/${{env.BUILD_CONFIGURATION}}
+      run: |
+        procdump64.exe -accepteula -i -r -ma c:/dumps/${{env.BUILD_PLATFORM}}/${{env.BUILD_CONFIGURATION}}
+
+    - name: Install LLVM and Clang
+      working-directory: ${{env.GITHUB_WORKSPACE}}
+      run: |
+        curl -fsSL -o LLVM10.exe https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/LLVM-10.0.0-win64.exe
+        7z x LLVM10.exe -y -o"C:/Program Files/LLVM"
+        echo "C:\Program Files\LLVM\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
 
     - name: Restore NuGet packages
       working-directory: ${{env.GITHUB_WORKSPACE}}
-      run: nuget restore ${{env.SOLUTION_FILE_PATH}}
+      run: |
+        cd external\ebpf-for-windows
+        nuget restore ${{env.EBPF_FOR_WINDOWS_SOLUTION_FILE}}
+        cd ..\..
```

Comment:
I am not sure you need the "cd ..\.." at the end of a step anywhere since each step has its own "working-directory" directive.